components/infobox/wikis/overwatch/infobox_map_custom.lua

```diff
@@ -18,69 +19,54 @@ local Flags = Lua.import('Module:Flags', {requireDevIfEnabled = true})
 local Widgets = require('Module:Infobox/Widget/All')
 local Cell = Widgets.Cell
 
-local CustomMap = Class.new()
-
+---@class OverwatchMapInfobox: MapInfobox
+local CustomMap = Class.new(Map)
 local CustomInjector = Class.new(Injector)
 
-local _args
-
 ---@param frame Frame
 ---@return Html
 function CustomMap.run(frame)
-	local customMap = Map(frame)
-	customMap.createWidgetInjector = CustomMap.createWidgetInjector
-	customMap.addToLpdb = CustomMap.addToLpdb
-	_args = customMap.args
-	return customMap:createInfobox()
-end
+	local map = CustomMap(frame)
+	map:setWidgetInjector(CustomInjector(map))
 
----@return WidgetInjector
-function CustomMap:createWidgetInjector()
-	return CustomInjector()
-end
-
----@param widgets Widget[]
----@return Widget[]
-function CustomInjector:addCustomCells(widgets)
-	local gameModes = CustomMap._getGameMode()
-	table.insert(widgets, Cell{
-		name = #gameModes == 1 and 'Game Mode' or 'Game Modes',
-		content = gameModes,
-	})
-	table.insert(widgets, Cell{
-		name = 'Lighting',
-		content = {_args.lighting}
-	})
-	table.insert(widgets, Cell{
-		name = 'Checkpoints',
-		content = {_args.checkpoints}
-	})
-	return widgets
+	return map:createInfobox()
 end
 
 ---@param id string
 ---@param widgets Widget[]
 ---@return Widget[]
 function CustomInjector:parse(id, widgets)
+	local args = self.caller.args
+	local gameModes = CustomMap._getGameMode(args)
```

Comment:
```suggestion
```

move it into the if
else it gets called vor every customizable
i.e. increases runtime with calling this several times for nothing